### DISCUSSION TITLE
Qute template changes leads to application reloads

### DISCUSSION THREAD

**Discussion by mr-mos** [quarkusio/quarkus#36670] [Q&A]
It this expected behavior?
Whenever I change my HTML template the whole application is reloaded when doing the browser-refresh.  That takes a little time and more annoying  deletes all session/cookie data....

From other frameworks, like Grails, template changes would apply without restarting the application.

Did I miss something with Quarkus and Qute?

Thanks!

[a comment by an automated account is not shown]

**Comment by mkouba** [comment]
Yes, it's intentional. In Qute we validate [type-safe expressions](https://quarkus.io/guides/qute-reference#typesafe_expressions), [type-safe templates](https://quarkus.io/guides/qute-reference#typesafe_templates) and some namespace expressions ([such as `cdi:`](https://quarkus.io/guides/qute-reference#injecting-beans-directly-in-templates)). In other words, there are many cases where we need to rebuild the app when a template is modified.

**Reply by mr-mos** [reply]
please see my post below

**Comment by mr-mos** [comment]
Hey Martin,
thanks for bringing some light into the darkness. I understand.  But it's really a kind of annoying that the application restarts for every little change (even just text).
In Quarkus you have the parameter "quarkus.live-reload.instrumentation=true" which even avoids restarts if you edit some class-files.

Could something similar be done in Qute?  Should I open an issue?   (Many other template-languages don't require restarting the server in development mode).

**Reply by mkouba** [reply]
> Could something similar be done in Qute? Should I open an issue? (Many other template-languages don't require restarting the server in development mode).

Hm, I'm not so sure we would be able to detect a change that would be safe, i.e. would not require the rebuild. The thing is that you could have a simple template like this:
```
Hello {name}!
```
And then turn this template into:
```
Hi {name}!
```
That is safe. However, if you do:
```
{@String name}
Hi {name}!
```
Then we need to rebuild the app and validate the `{name}`.

Maybe we should add a config option that would allow users to define the "development safe" templates, i.e. templates that do not require app restart... it's not perfect but might come in handy.

**Reply by mr-mos** [reply]
That sounds like a good compromise. Should I create a ticket for this?

**Reply by mkouba** [reply]
Yes please.

**Comment by mr-mos** [comment]
Issue created as discussed: https://github.com/quarkusio/quarkus/issues/36692
